aea/configurations/data_types.py

```diff
@@ -452,6 +455,122 @@ def __lt__(self, other: Any) -> bool:
         )
 
 
+class ExtendedPublicId(PublicId):
```

Comment:
To use hashes as a part of the PublicId, right now it follows the following format

`author/package:version:hash`

But I think we'll need to come with a better format because I think this issue might lead to design issues

Reply:
@angrybayblade what's a better format suggestion you have?

Reply:
So right now we can work this. I'll try to come up with a better format when performing refactors on this repo